pkg/bootflow/types/trust_chain.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type TrustChains map[reflect.Type]TrustChain
+
+type TrustChain interface {
```

Comment:
Could you please provide a couple of examples?
It looks like that we need more methods, to at least check for error

And why do we need IsInitialized?

Reply:
Found examples, sorry :)

Reply:
> Could you please provide a couple of examples?

Here is an example of using a `TrustChain`:
https://github.com/9elements/converged-security-suite/blob/60c0e0b6aa636851b4a7c0a398d47202383705d8/pkg/bootflow/examples/ocppeipcr0/main.go

There you can find:
```go
	tpmInstance, err := tpm.GetFrom(state)
	if err != nil {
		panic(err)
	}
```

It extracts `TrustChain` with type `*TPM` from the `state`.

> It looks like that we need more methods, to at least check for error

Unfortunately, different `TrustChain`-s may have different designs (for example DICE and TPM measurements) and because of that we do not have common methods.
But yet we need an interface we will use for the `map` of all `TrustChain`-s.

> And why do we need `IsInitialized`?

That was the only common method I thought of for different `TrustChain`-s. So two reasons:
1. One may use it to know if this `TrustChain` was initialized during a given boot process. If are more common methods for any abstract `TrustChain`, I'd gladly add them here to unlock more opportunities.
2. To do not have `interface{}` and thus to enforce compile time checks that a good type is provided in an argument to somewhere.